### PR TITLE
Add styletron-react types

### PR DESCRIPTION
**Pull request by erictaylor**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

This requires `@types/styletron-standard` #35301 be merged and released.

### CODE REVIEW

File: types/styletron-react/index.d.ts
Comment (erictaylor) [inline types/styletron-react/index.d.ts]: Is it okay to re-export types from `styletron-standard` here? Or should this be avoided in favor of using the types directly for their defs?

File: types/styletron-react/package.json
Comment (erictaylor) [inline types/styletron-react/package.json]: Is this the right way of doing this? This package is currently in #35301. I'm not sure what the initial version will be, but I assume 1.0.0?

File: types/styletron-react/index.d.ts
Comment (erictaylor) [inline types/styletron-react/index.d.ts]: Is there a way to directly use the `WithTransformFn` interface when describing this function? Instead of basically duplicating the typing?

**Comment by erictaylor** [comment]
Tests should pass once `styletron-standard` is merged and `master` is merged into this branch.

File: types/styletron-react/index.d.ts
Comment (sandersn) [reply types/styletron-react/index.d.ts]: I think it is fine for convenience. Maybe the maintainers of styletron-standard have an opinion?

File: types/styletron-react/index.d.ts
Comment (sandersn) [reply types/styletron-react/index.d.ts]: probably `export const withTransform: WithTransformFn`

**Review by sandersn — APPROVED** [review]
This looks fine to me. Do you want to change `withTransform` before I merge it?

**Comment by erictaylor** [comment]
> This looks fine to me. Do you want to change `withTransform` before I merge it?

@sandersn adcb0af refactors `withTransform`, `withWrapper` and `withStyleDeep`. We should be good to go for a merge.

**Pull request merged**
